File: binary.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1  # If the target is not found in the array

# Example usage:
arr = [2, 3, 4, 10, 40]
target = 10
result = binary_search(arr, target)
if result != -1:
    print(f"Element is present at index {result}")
else:
    print("Element is not present in the array")

File: Program's_Contributed_By_Contributors/C++ Programs/searching/binary.cpp
#include <iostream>
using namespace std;

int binary(int arr[], int key, int start, int end) {
    int mid;
    while (start <= end) {
        mid = (start + end) / 2;
        if (arr[mid] == key)
            return mid;
        else if (key > arr[mid])
            start = mid + 1;
        else
            end = mid - 1;
    }
    return -1;
}

int Recursive(int arr[], int key, int start, int end) {
    if (start <= end) {
        int mid = (start + end) / 2;
        if (arr[mid] == key)
            return mid;
        else if (key > arr[mid])
            return Recursive(arr, key, mid + 1, end);
        else
            return Recursive(arr, key, start, mid - 1);
    }
    return -1;
}

int main() {
    int arr[] = {10, 20, 30, 50, 60, 80, 100, 110, 130, 170};
    int n = sizeof(arr) / sizeof(arr[0]);

    int key = 100;
    int result = binary(arr, key, 0, n - 1);
    (result == -1)
        ? cout << "Element " << key << " not found"
        : cout << "Element " << key << " found at index " << result;
    cout << endl;
    
    key = 20;
    result = Recursive(arr, key, 0, n - 1);
    (result == -1)
        ? cout << "Element " << key << " not found"
        : cout << "Element " << key << " found at index " << result;
}